src/components/ec-full-screen-overlay/ec-full-screen-overlay.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <transition name="ec-full-screen-overlay__fade">
+    <div
+      class="ec-full-screen-overlay"
+      data-test="ec-full-screen-overlay"
+    >
+      <div
+        ref="overlayContainer"
+        class="tw-col-full"
+        data-test="ec-full-screen-overlay__container"
+      >
+        <div class="tw-grid tw-grid-cols-3">
+          <div class="tw-col-full sm:tw-col-3" />
+          <div class="tw-col-full sm:tw-col-6">
+            <div class="tw-grid">
+              <div class="tw-col-full sm:tw-col-9">
+                <h1
+                  v-if="title"
+                  data-test="ec-full-screen-overlay__title"
+                  class="ec-full-screen-overlay__title"
+                >
+                  {{ title }}
+                </h1>
+              </div>
+              <div class="tw-col-full sm:tw-col-3 tw-text-right">
+                <a
+                  href="#"
+                  data-test="ec-full-screen-overlay__close-icon-container"
+                  @click="emit('closeOverlay')"
```

Comment:
Emits should be in `kebab-case` but there's no need to repeat the name of the component in it, so `close` is enough.